File: JavaScript/Tyres.js
import Car from "./Car.js";

// Esta clase representa los neumáticos del carro
export default class Tyres extends Car {
  // Constructor de los neumáticos
  constructor(A, B, C, D, lapsMax, isWet, tyreColor) {
    super(A, B, C, D);
    this.durabilityMax = lapsMax;
    this.wet = isWet;
    this.color = tyreColor;
  }

  // Método de durabilidad base (puede sobreescribirse)
  Durability(laps) {
    return `the tyres have been on track ${laps} laps`;
  }

  // Mostrar toda la información del neumático
  ShowAll() {
    console.log(
      `The tyres are: ${this.rightFront}, ${this.leftFront}, ${this.rightRear}, ${this.leftRear}\nTheir durability is ${this.durabilityMax} laps\nThe tyres are wet? ${this.wet}\nTyres color: ${this.color}`
    );
  }

  // Método para la presión (a implementar por cada compuesto)
  PSI() {
    console.log("NON");
  }
}

File: Go/Tyres.go
package main
import "fmt"

type Tyres struct{
	Car
	durabilityMax int
	wet bool
	color string
}

func NewTyres(A, B, C, D string, lapsMax int, isWet bool, tyreColor string) Tyres {
	car := Car{
		rightFront: A,
		leftFront:  B,
		rightRear:  C,
		leftRear:   D,
	}
	return Tyres{
		Car:           car,
		durabilityMax: lapsMax,
		wet:           isWet,
		color:         tyreColor,
	}
}

func (t Tyres) Durability(laps int) string {
	return fmt.Sprintf("The tyres have been on track %d laps", laps)
}

func (t Tyres) ShowAll(){
	fmt.Printf("The tyres are %s %s %s %s\nTheir durability is %d laps\nThe tyres are wet? %t\nTyres color: %s\n",
		t.rightFront, t.leftFront, t.rightRear, t.leftRear, t.durabilityMax, t.wet, t.color)
}

func (t Tyres) PSI(){}